src/shapes/itext.class.ts

```diff
@@ -166,10 +167,14 @@ export class IText extends ITextClickBehaviorMixin<ITextEvents> {
   _set(key: string, value: any) {
     if (this.isEditing && this._savedProps && key in this._savedProps) {
       this._savedProps[key] = value;
-    } else {
-      super._set(key, value);
+      return this;
+    }
+    if (key === 'canvas') {
+      this.canvas instanceof Canvas &&
+        this.canvas.textEditingManager.remove(this);
+      value instanceof Canvas && value.textEditingManager.add(this);
```

Comment:
Is his the correct place to catch this?
Can't we leverage the collection onObjectAdded/onObjectRemoved to handle this?

Reply:
it is not good enough taking into account nested targets

Reply:
What does it mean nested targets?

Reply:
yes, text under groups